fix-android-dependencies.py

```diff
@@ -0,0 +1,109 @@
+COMPILE_SDK_VERSION = 27
+TARGET_SDK_VERSION = 27
+BUILD_TOOLS_VERSION = '27.0.2'
+
+COMPILE_SDK_RE = r'compileSdkVersion[\s][\w]+'
+TARGET_SDK_RE = r'targetSdkVersion[\s][\w]+'
+BUILD_TOOLS_RE = r'buildToolsVersion[\s][\'\"][\w\.]+[\'\"]'
+
+GROUPS_ALLOWED_MAJOR_UPDATE = [
+  'com.google.android.gms',
+  'com.android.support'
+]
+
+# Depends on https://github.com/ben-manes/gradle-versions-plugin
+#
+# Must run this command:
+# $ ./gradlew dependencyUpdates -Drevision=release -DoutputFormatter=json
+INPUT_JSON = 'build/dependencyUpdates/report.json'
+
+def findGradleFiles():
+    """Finds all build.gradle files, recursively."""
+    gradle_files = []
+    for root, dirs, files in os.walk('.'):
+        for file in files:
+            if file.endswith('build.gradle'):
+                gradle_files.append(os.path.join(root, file))
+
+    return gradle_files
+
+def getAndroidReplacements():
+    """Gets a dictionary of all android-specific replacements to be made."""
+    replacements = {}
+
+    compileSdk = 'compileSdkVersion {}'.format(COMPILE_SDK_VERSION)
+    targetSdk = 'targetSdkVersion {}'.format(TARGET_SDK_VERSION)
+    buildToolsVersion = 'buildToolsVersion \'{}\''.format(BUILD_TOOLS_VERSION)
+
+    replacements[COMPILE_SDK_RE] = compileSdk
+    replacements[TARGET_SDK_RE] = targetSdk
+    replacements[BUILD_TOOLS_RE] = buildToolsVersion
+
+    return replacements
+
+def isMajorUpdate(old_version, new_version):
+  """Compares version strings to see if it's a major update."""
+  old_major = old_version.split('.')[0]
+  new_major = new_version.split('.')[0]
+
+  return old_major != new_major
+
+def getDepReplacements():
+    """Gets a dictionary of all dependency replacements to be made."""
+    replacements = {}
+    with open(INPUT_JSON, 'r') as f:
+        json_data = json.loads(f.read())
+
+        outdated_deps = json_data['outdated']['dependencies']
+        for dep in outdated_deps:
+            group = dep['group']
+            name = dep['name']
+
+            curr_version = dep['version']
+            new_version = dep['available']['release']
+
+            if (isMajorUpdate(curr_version, new_version) and
+                group not in GROUPS_ALLOWED_MAJOR_UPDATE):
+              print 'Skipping major update to {}:{}'.format(group, name)
+              continue
+
+            curr_dep = '{}:{}:{}'.format(group, name, curr_version)
+            new_dep = '{}:{}:{}'.format(group, name, new_version)
+            replacements[curr_dep] = new_dep
+
+    return replacements
+
+def updateAll():
+    """Runs through all build.gradle files and performs replacements."""
+    replacements = {}
+    replacements.update(getAndroidReplacements())
+    replacements.update(getDepReplacements())
+
+    # Print all updates found
+    print 'Dependency updates:'
+    for (k, v) in replacements.iteritems():
+        print '{} --> {}'.format(k, v)
+
+    # Iterate through each file and replace it
+    for file in findGradleFiles():
```

Comment:
`file` is a built-in, please don't shadow it (suggestion: use `grade_file` or `filename` instead.